File: 1103/main.go
package main

import "fmt"

func main() {
	fmt.Println(distributeCandies(7, 4))	
	fmt.Println(distributeCandies(10, 3))	
	fmt.Println(distributeCandies(60, 4))	// [15,18,15,12]
}

func distributeCandies(candies int, num_people int) []int {
	ans := make([]int, num_people)
	t := 0
	for candies > 0 {
		for i := 1; i <= num_people; i++ {
			t++
			if candies <= t {
				ans[i-1] = ans[i-1] + candies
				candies = 0
				break
			} else {
				ans[i-1] = ans[i-1] + t
				candies -= t
			}

		}
	}

	return ans
}

File: 1103/main.rs
use std::convert::TryInto;

struct Solution{}

impl Solution {
    pub fn distribute_candies(candies: i32, num_people: i32) -> Vec<i32> {
        let mut ans = vec![0;num_people.try_into().unwrap()];
        let mut t = 0;
        let mut c = candies;
        while c > 0 {
            for j in 1..num_people+1 {
                let i = j as usize;
                t += 1;
                if c <= t {
                    ans[i-1] = ans[i-1] + c;
                    c = 0;
                    break
                } else {
                    ans[i-1] = ans[i-1] + t;
                    c -= t;
                }
            }
        }
        ans
    }
}

fn main() {
    println!("{:#?}", Solution::distribute_candies(60, 4))  // [15,18,15,12]
}